### DISCUSSION TITLE
Improved Theming Support

### DISCUSSION THREAD

**Discussion by someone13574** [zed-industries/zed#23956] [Feature Requests]
Ability for themes to change more parts of the ui other than just color, such as border width, margin, padding, and border radius.

**Comment by MrDrMcCoy** [comment]
Seconded. It would be especially great to be able to reduce the padding around line numbers.

**Comment by maxdeviant** [comment]
This probably needs more discussion.

**Comment by Sutaai** [comment]
This would, overall, allow more control, freedom, customization, pretty much open up imagination, and I'd love to see it implemented, however, I may not entirely wish either.

- Hard to maintain and will likely break with time, not even mentioning it's a new coding rule for underlying developers of Zed
- Will undeniably end up being against the original concept of Zed's design language
- Require additional documentation from Zed
- Potential performance degradation

It's a difficult issue to tackle. Is the maintenance worth the feature?
